### PR TITLE
dataset-items

### PR DESCRIPTION
**Pull request by nenadnovovic-ab**
## Summary by CodeRabbit

- **New Features**
	- Introduced methods to create, delete, and update dataset items, enhancing dataset management capabilities.

- **Tests**
	- Added test cases for the new dataset item management methods to ensure expected behavior and reliability.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -424,4 +461,35 @@ export class AutoblocksAPIClient {
...
+    return this.delete(
+      `/datasets/${encodeURIComponent(args.datasetExternalId)}/items/${encodeURIComponent(args.itemId)}`,
+    );
+  }
+
+  public async updateDatasetItem(args: {
+    datasetExternalId: string;
+    itemId: string;
+    data: Record<string, unknown>;
+    splitNames: string[];
```
[the remark this answers is not in the document]
Comment (adamnolte) [reply src/client.ts:488]: I think we should just add split_names to the create method

[the remark this answers is not in the document]
Comment (nenadnovovic-ab) [reply src/client.ts:488]: @adamnolte but we can't add splits when creating an item, only on update?
the same logic is in the webapp

File: src/client.ts [hunk 2]
```diff
@@ -424,4 +461,35 @@ export class AutoblocksAPIClient {
...
+    datasetExternalId: string;
+    itemId: string;
+  }): Promise<{ id: string }> {
+    return this.delete(
+      `/datasets/${encodeURIComponent(args.datasetExternalId)}/items/${encodeURIComponent(args.itemId)}`,
+    );
+  }
+
+  public async updateDatasetItem(args: {
+    datasetExternalId: string;
```
Comment (adamnolte) [inline src/client.ts:485]: lets call this name so its consistent with getDataset. externalId is really an internal naming convention

File: src/client.ts [hunk 3]
```diff
@@ -424,4 +461,35 @@ export class AutoblocksAPIClient {
...
+    data: Record<string, unknown>;
+  }): Promise<{ id: string }> {
+    return this.post(
+      `/datasets/${encodeURIComponent(args.datasetExternalId)}/items`,
+      { data: args.data },
+    );
+  }
+
+  public async deleteDatasetItem(args: {
+    datasetExternalId: string;
```
Comment (adamnolte) [inline src/client.ts:476]:
```suggestion
    name: string;
```

File: src/client.ts [hunk 4]
```diff
@@ -424,4 +461,35 @@ export class AutoblocksAPIClient {
       `/datasets/${encodedName}/schema-versions/${encodedSchemaVersion}${splitsQueryParam}`,
     );
   }
+
+  public async createDatasetItem(args: {
+    datasetExternalId: string;
```
Comment (adamnolte) [inline src/client.ts:466]:
```suggestion
    name: string;
```

File: src/client.ts (hunk 1 above)
Comment (adamnolte) [inline src/client.ts:488]:
```suggestion
    splits: string[];
```

File: src/client.ts [hunk 5]
```diff
@@ -424,4 +461,35 @@ export class AutoblocksAPIClient {
...
+      `/datasets/${encodeURIComponent(args.datasetExternalId)}/items/${encodeURIComponent(args.itemId)}`,
+    );
+  }
+
+  public async updateDatasetItem(args: {
+    datasetExternalId: string;
+    itemId: string;
+    data: Record<string, unknown>;
+    splitNames: string[];
+  }): Promise<{ id: string }> {
```
Comment (adamnolte) [inline src/client.ts:489]: is this returning the revision id?

Comment (nenadnovovic-ab) [reply src/client.ts:490]: yes, in all three methods: 
` return c.json({ id: newRevision.id });`

Comment (adamnolte) [reply src/client.ts:490]: Let's call it revisionId here

**Review by adamnolte — APPROVED** [review]

**Pull request merged**